File: scribbleshare-commons/src/main/java/net/stzups/board/data/objects/canvas/object/objects/Shape.java
package net.stzups.board.data.objects.canvas.object.objects;

import io.netty.buffer.ByteBuf;
import net.stzups.board.data.objects.canvas.object.CanvasObject;

public class Shape extends CanvasObject {
    private final int width;
    private final int height;

    public Shape(ByteBuf byteBuf) {
        super(byteBuf);
        width = byteBuf.readUnsignedShort();
        height = byteBuf.readUnsignedShort();
    }

    public void serialize(ByteBuf byteBuf) {
        super.serialize(byteBuf);
        byteBuf.writeShort((short) width);
        byteBuf.writeShort((short) height);
    }
}

File: scribbleshare-frontend/src/scripts/canvas/canvasObjects/Shape.js
import CanvasObject from "../CanvasObject.js";
import {ctx} from "../Canvas.js";

export default class Shape extends CanvasObject {
    constructor(reader) {
        super(reader);
    }

    draw() {
        ctx.fillRect(0, 0, this.width, this.height);
    }

    serialize(writer) {
        super.serialize(writer);
    }

    static create(x, y, width, height) {
        let shape = Object.create(this.prototype);
        shape.x = x;
        shape.y = y;
        shape.width = width;
        shape.height = height;
        shape.rotation = 0;
        return shape;
    }
}